javascript-bank/account.js

```diff
@@ -0,0 +1,39 @@
+/* exported Account */
+function Account(number, holder){
+  this.number = number;
+  this.holder = holder;
+  this.transactions = [];
+  this.balance = 0;
+}
+
+Account.prototype.deposit = function (amt) {
+  if ((amt > 0) && ((amt % 1) === 0)) {
+    let trans = new Transaction('deposit', amt);
+    this.transactions.push(trans);
+    this.balance += amt;
+    this.number += amt
+    return true
+  } else {
+    return false
+  }
+};
+
+Account.prototype.withdraw = function (amt) {
+  if ((amt > 0) && ((amt % 1) === 0)) {
+    let trans = new Transaction('withdrawal', amt);
+    this.transactions.push(trans);
+    this.balance -= amt;
+    this.number -= amt;
```

Comment:
The `number` property is a unique account number identifier and should not change for the life of the account.

---

![image](https://user-images.githubusercontent.com/16143771/123365534-4a317f80-d52b-11eb-9b25-dbd832bccb31.png)